framework/arcane-framework/src/main/scala/services/streaming/StreamGraphBuilder.scala

```diff
@@ -0,0 +1,56 @@
+package com.sneaksanddata.arcane.framework
+package services.streaming
+
+import models.DataCell
+import services.mssql.MsSqlConnection
+import services.mssql.MsSqlConnection.VersionedBatch
+import services.mssql.query.LazyQueryResult.OutputType
+import services.mssql.query.QueryRunner
+import services.streaming.base.{HasVersion, StreamLifetimeService}
+
+import zio.stream.ZStream
+import zio.{Task, ZIO}
+
+import java.time.Duration
+import scala.util.{Failure, Try}
+
+
+given HasVersion[VersionedBatch] with
+  extension (result: VersionedBatch)
+    def getLatestVersion: Option[Long] =
+      val (queryResult, version) = result
+
+      // If the database response is empty, we can't extract the version from it and return the old version.
+      queryResult.read.headOption match
+        case None => Some(version)
+        case Some(row) =>
+          // If the database response is not empty, we can extract the version from any row of the response.
+          // Let's take the first row and try to extract the version from it.
+          val dataVersion = row.filter(_.name == "ChangeTrackingVersion") match
+            // For logging purposes will be used in the future.
+            case Nil => Failure(new UnsupportedOperationException("No ChangeTrackingVersion found in row."))
+            case version :: _ => Try(version.value.asInstanceOf[Long])
+          dataVersion.toOption
+
+
+
+class StreamGraphBuilder(msSqlConnection: MsSqlConnection) {
```

Comment:
ideally this should be implicit val so it can't be changed or provided from user code outside of main

Reply:
TBD later